src/Microsoft.AspNet.Razor/Generator/TagHelperCodeGenerator.cs

```diff
@@ -66,6 +66,7 @@ public override void GenerateStartBlockCode(Block target, CodeGeneratorContext c
 
                 attributes[attribute.Key] = new ChunkBlock
                 {
+                    Association = first?.Association,
```

Comment:
OIC  `target` won't work because it's the whole HTML element / `TagHelperBlock` and starts prior to the attribute value.  think of cases like
``` html
  <tag-helper
    some-attribute="some-value"
    another-attribute="another-value" />
```

(suspect I can change VS options to do the above by default :smiley_cat:)